pkg/controllers/logset/bucketclaim.go

```diff
@@ -0,0 +1,128 @@
+// Copyright 2023 Matrix Origin
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package logset
+
+import (
+	"fmt"
+	recon "github.com/matrixorigin/controller-runtime/pkg/reconciler"
+	"github.com/matrixorigin/matrixone-operator/api/core/v1alpha1"
+	kruisev1 "github.com/openkruise/kruise-api/apps/v1beta1"
+	"k8s.io/apimachinery/pkg/api/equality"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sort"
+)
+
+func (r *Actor) syncBucketClaim(ctx *recon.Context[*v1alpha1.LogSet], sts *kruisev1.StatefulSet) error {
+	ls := ctx.Obj
+
+	// skip if no s3 share storage configuration
+	if ctx.Obj.Spec.LogSetBasic.SharedStorage.S3 == nil {
+		return nil
+	}
+
+	bucket, err := v1alpha1.ClaimedBucket(ctx.Client, ctx.Obj)
+	if err != nil {
+		return err
+	}
+
+	// create new bucket if no bucket found
+	if bucket == nil {
+		return r.createNewBucket(ctx, sts)
```

Comment:
I think we should continue here rather than return, or return a resync error. Otherwise `syncBucketClaim` is not guaranteed to be called again